sdk/tests/utilities/test_api_client_factory_auth_flow.py

```diff
@@ -0,0 +1,63 @@
+import unittest
+from lusid import InstrumentsApi, ResourceListOfInstrumentIdTypeDescriptor
+from lusid.utilities import ApiClientFactory
+from utilities import CredentialsSource
+from unittest.mock import patch
+
+source_config_details, config_keys = CredentialsSource.fetch_credentials(), CredentialsSource.fetch_config_keys()
+
+
+class ApiFactory(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # create a configured API client
+        cls.secrets = CredentialsSource.secrets_path()
+
+    def validate_api(self, api):
+        result = api.get_instrument_identifier_types()
+        self.assertIsNotNone(result)
+        self.assertIsInstance(result, ResourceListOfInstrumentIdTypeDescriptor)
+        self.assertGreater(len(result.values), 0)
+
+    def test_get_api_with_access_token_env_var(self):
```

Comment:
mock out the env vars here (this test has an assumption they have been set), see https://github.com/finbourne/lusid-sdk-python-preview/blob/master/sdk/tests/utilities/test_api_client_builder.py#L211 for an example on how to do this